src/main/java/org/sagebionetworks/web/client/widget/modal/Dialog.java

```diff
@@ -127,4 +155,32 @@ public void hide() {
 	public void addStyleName(String style) {
 		modal.addStyleName(style);
 	}
+	
+	public Widget asWidget() {
+		return widget;
+	}
+
+	public boolean isVisible() {
+		return widget.isVisible();
+	}
+
+	@Override
+	public void add(Widget w) {
```

Comment:
When you implement HasWidgets, I'm *fairly certain* that when you add contents to the container in the .ui.xml, it looks to this method to know where to add the contents.